File: ISL.ReIdentification.Core/Models/Orchestrations/Accesses/AccessRequest.cs
﻿// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using ISL.ReIdentification.Core.Models.Foundations.CsvIdentificationRequests;
using ISL.ReIdentification.Core.Models.Foundations.ImpersonationContexts;
using ISL.ReIdentification.Core.Models.Foundations.ReIdentifications;

namespace ISL.ReIdentification.Core.Models.Orchestrations.Accesses
{
    public class AccessRequest
    {
        public IdentificationRequest IdentificationRequest { get; set; }
        public CsvIdentificationRequest CsvIdentificationRequest { get; set; }
        public ImpersonationContext ImpersonationContext { get; set; }
    }
}

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Client/src/models/accessRequest/accessRequest.ts
import { CsvIdentificationRequest } from "../csvIdentificationRequest/csvIdentificationRequest";
import { ImpersonationContext } from "../impersonationContext/impersonationContext";
import { IdentificationRequest } from "../ReIdentifications/IdentificationRequest";

export class AccessRequest {
    public identificationRequest?: IdentificationRequest;
    public csvIdentificationRequest?: CsvIdentificationRequest;
    public impersonationContext?: ImpersonationContext;

    constructor(accessRequest: AccessRequest) {
        this.identificationRequest = accessRequest.identificationRequest || new IdentificationRequest();
        this.csvIdentificationRequest = accessRequest.csvIdentificationRequest || {};
        this.impersonationContext = accessRequest.impersonationContext || new ImpersonationContext();
    }
}